bot/controllers/roles.js

```diff
@@ -1,119 +1,176 @@
-    if (msg.parsed[0].toLowerCase() === '!addallroles') {
-      if (!message.guild) return noGuildFault(message);
-      message.guild.roles.map((role) => {
-        if (!disallowedRoles.includes(role.name.toLowerCase())) {
-          return message.member.addRole(role).catch(util.log);
+    const onSuccess = (message, res, ctrl) => {
+      if (res !== null) {
+        if (ctrl.resType === 'reply') {
+          return message.reply(res);
+        } else if (ctrl.resType === 'dm') {
+          return message.author.send(res);
         }
-        return role.name;
-      });
+      }
+      // Fail Safe
+      return false;
+    };
 
-      return message.reply('Adding you to all Roles. You\'re going to be drinking from the firehose :sob:');
-    }
-    return false;
-  };
+    const onError = message => message.reply('I Broke... Beep...Boop...Beep');
 
-  const _removeAllRoles = (message) => {
-    const msg = messageMiddleware(message);
-    if (msg.parsed[0].toLowerCase() === '!removeallroles') {
-      if (!message.guild) return noGuildFault(message);
-      message.guild.roles.map((role) => {
-        if (!disallowedRoles.includes(role.name.toLowerCase())) {
-          return message.member.removeRole(role).catch(util.log);
-        }
-        return role.name;
-      });
+    const messageMiddleware = (message) => {
+      const container = {};
+      container.parsed = message.content.split(' ');
+      const msg = Object.assign(message, container);
+      return msg;
+    };
 
-      return message.reply('Removing all roles. Back to basics.');
-    }
-    return false;
-  };
+    ctrls.map((ctrl) => {
+      util.log('Running through controller', ctrl.cmd, 2);
 
-  const _addRole = (message) => {
-    util.log('passed', message.content);
-    const msg = messageMiddleware(message);
-    if (msg.parsed[0].toLowerCase() === '!addrole') {
-      // TODO: Reduce Code Duplication
-      if (!message.guild) return noGuildFault(message);
-      const targetRole = message.guild.roles.find('name', msg.parsed[1]);
-      if (targetRole === null) {
-        return message.reply('"' + msg.parsed[1] + '" is not a known role. Try `!roles` to get a list of all Roles (They are case-sensitive)');
-      }
-      if (disallowedRoles.includes(targetRole.name.toLowerCase())) {
-        return message.reply('You are not worthy.');
-      }
+      const msg = messageMiddleware(message);
+      if (msg.parsed[0].toLowerCase() === ctrl.cmd.toLowerCase()) {
+        util.log('!!! Matched Ctrl to Cmd !!!', ctrl.cmd, 2);
 
-      // TODO: Handle error to respond with message
-      // TODO: Change catch to pass to util.error... will need created
-      message.member.addRole(targetRole).catch(util.log);
-      // TODO: Create verbose response toggle?
-      // message.reply(targetRole.name + ' added.');
-      return true;
-    }
-    return false;
-  };
+        // Ensure the communication is happening on a server
+        if (!ctrl.alllowInDM) {
```

Comment:
Typo?

Reply:
Careful with feedback this ambiguous. It took me awhile double and triple checking the logic thinking that you were referencing a logical typo. But now I see the triple ```lll``` misspelling.

Great catch though!

Reply:
Aha, sorry about that. It was super obvious to my eyes but I can see how it might not be to everyone. Will remember that in the future.